ceramic-ahash/src/recon/mod.rs

```diff
@@ -0,0 +1,255 @@
+use crate::AHash;
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+use std::fmt::Display;
+use std::ops::Bound::Excluded;
+
+#[cfg(test)]
+pub mod tests;
+
+// Recon
+// short for reconnaissance
+// short for set reconciliation based synchronization
+//
+// A recon should hold the list of keys and generate responses
+// when processing messages
+//
+#[derive(Debug)]
+pub struct Recon {
+    pub keys: BTreeMap<String, AHash>, // this will be a b#tree at some point in the future
+}
+
+impl Recon {
+    pub fn from_set(s: BTreeSet<String>) -> Recon {
+        let mut r = Recon {
+            keys: BTreeMap::default(),
+        };
+        for key in s {
+            r.insert(&key);
+        }
+        r
+    }
+
+    pub fn insert(&mut self, key: &str) {
+        self.keys.insert(key.to_string(), AHash::digest(key));
+    }
+
+    pub fn hash_range<H: Hash>(&self, left_fencepost: &str, right_fencepost: &str) -> H {
```

Comment:
does this function need to be public? I don't see any public uses and it feels like an internal detail of the Recon implementation.